KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,652 @@
+
+## Motivation
+
+<!--The motivation is critical for KIPs that want to change the Klaytn protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the KIP solves. KIP submissions without sufficient motivation may be rejected outright.-->
+
+Transparency is the one of most important aspect of blockchain. It is important to ensure that treasury funds are allocated and managed in a transparent and verifiable manner. The proposed smart contract aims to disclose the management of treasury funds in a transparent manner through smart contracts reducing the risk of errors and mismanagement. By providing transparency and accountability in the allocation and management of funds, the smart contract can help to build trust and confidence among stakeholders.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Klaytn platforms (klaytn). -->
+
+The proposed smart contract will be implemented in Solidity and will be compatible with the Ethereum Virtual Machine (EVM). The smart contract will use Ownable contract to restrict access to certain functions to the owner of the contract.
+
+The smart contract will have the following features:
+
+- Add/Remove fund addresses
+- Approve fund addresses and fund allocation
+- Reset the storage values at any unforeseen circumstances before Finalized
+- Finalize the smart contract after execution
+
+### Smart Contracts Overview
+
+#### Enums
+
+The smart contract will have the following enum to track the status of the contract:
+
+- `Initialized - 0`: The initial state of the contract.
+- `Registered - 1`: Senders and receivers registered.
+- `Approved - 2`: Senders approved.
+- `Finalized - 3`: Rebalance executed and finalized.
+
+#### Life Cycle
+
+The contract status should follow the ENUM order above during status change. The only way to go to previous state is by calling Reset() function.
+
+![](../assets/kip-103/lifecycle.png)
+
+Status transition
+
+- Initialized → Registered → Approved → Finalized ✅
+- Initialized → Registered → Finalized ❌
+- Initialized → Approved ❌
+- Initialized → Finalized ❌
+
+#### Structs
+
+The smart contract will have the following structs :
+
+- `Retired`: to store the details of retired and approver addresses.
+- `Newbie`: to store newbie and amount.
+
+#### Storage
+
+The smart contract will have the following storage variables:
+
+- `retirees`: array of retirees structs.
+- `newbie`: array of newbies structs.
+- `status`: current status of the contract.
+- `rebalanceBlockNumber`: Block number of the execution of rebalancing.
+- `memo`: result of the treasury fund rebalance.
```

Comment:
What does the memo look like? it seems a string, then we can specify some strings of this memo.